internal/provider/cluster_resource_test.go

```diff
@@ -828,6 +828,23 @@ func TestIntegrationServerlessClusterResource(t *testing.T) {
 		finalCluster      client.Cluster
 		ignoreImportPaths []string
 	}{
+		{
+			name: "serverless rejects BYOC in Create",
+			createStep: func() resource.TestStep {
+				return resource.TestStep{
+					Config: fmt.Sprintf(`
+                        resource "cockroach_cluster" "test" {
+                            name           = "%s"
+                            cloud_provider = "GCP"
+                            serverless = {}
+                            regions = [{ name = "us-central1" }]
+                            customer_cloud_account = { gcp = { service_account_email = "sa@example.iam.gserviceaccount.com" } }
+                        }
+                    `, clusterName),
+					ExpectError: regexp.MustCompile(`These attributes cannot be configured together:\s\[serverless,customer_cloud_account\]`),
```

Comment:
note: this looks like the error message from the  automatic checking based on the resource validator

Reply:
Ya that's what it is. Seems like a reasonable test to make sure this doesn't change by accident in the future.